apps/cowswap-frontend/src/modules/permit/hooks/useIsTokenPermittable.ts

```diff
@@ -21,17 +24,22 @@ import { checkIsTokenPermittable } from '../utils/checkIsTokenPermittable'
  * When it is not, returned type is `false`
  * When it is unknown, returned type is `undefined`
  *
- * @param token
  */
-export function useIsTokenPermittable(token: Nullish<Currency>): IsTokenPermittableResult {
+export function useIsTokenPermittable(
+  token: Nullish<Currency>,
+  tradeType: Nullish<TradeType>
+): IsTokenPermittableResult {
   const { chainId } = useWalletInfo()
   const { provider } = useWeb3React()
 
   const lowerCaseAddress = token?.wrapped?.address?.toLowerCase()
   const isNative = token?.isNative
   const tokenName = token?.name || lowerCaseAddress || ''
 
-  const isPermitEnabled = useIsPermitEnabled(chainId)
+  // Avoid building permit info in the first place if order type is not supported
+  const isPermitSupported = !!tradeType && ORDER_TYPE_SUPPORTS_PERMIT[tradeType]
+
+  const isPermitEnabled = useIsPermitEnabled(chainId) && isPermitSupported
```

Comment:
The main change, disabling permit check completely if order type is not supported